onnxruntime/core/session/abi_session_options_impl.h

```diff
@@ -14,6 +14,7 @@ struct OrtSessionOptions {
   onnxruntime::SessionOptions value;
   std::vector<OrtCustomOpDomain*> custom_op_domains_;
   std::vector<std::shared_ptr<onnxruntime::IExecutionProviderFactory>> provider_factories;
+  std::vector<std::shared_ptr<void>> custom_op_lib_handles_;
```

Comment:
>std::vector<std::shared_ptr<void>> custom_op_lib_handles_; [](http://example.com/codeflow?start=2&length=58)

I suggest we use std::shared_ptr<vector>.
That way the whole vector becomes optional.